core/templates/pages/contributor-dashboard-admin-page/contributor-dashboard-tables/contributor-admin-stats-table.component.spec.ts

```diff
@@ -313,4 +334,195 @@ describe('Contributor stats component', () => {
         } as QuestionReviewerStatsData));
     }));
   });
+
+  describe('when toggled various model', () => {
+    beforeEach(() => {
+    });
+
+    it('should open question role editor modal and return not' +
+      ' translation submitter', fakeAsync(() => {
+      const removeRightsSpy = spyOn(
+        contributorDashboardAdminBackendApiService,
+        'removeContributionReviewerAsync');
+      const changes: SimpleChanges = {
+        activeTab: {
+          currentValue: component.TAB_NAME_QUESTION_SUBMITTER,
+          previousValue: null,
+          firstChange: true,
+          isFirstChange: () => true,
+        },
+      };
+      component.activeTab = component.TAB_NAME_QUESTION_SUBMITTER;
+      component.ngOnChanges(changes);
+      spyOn(
+        contributorDashboardAdminBackendApiService,
+        'contributionReviewerRightsAsync').and.returnValue(Promise.resolve({
+        can_submit_questions: true,
+        can_review_questions: true,
+        can_review_translation_for_language_codes: [],
+        can_review_voiceover_for_language_codes: []
+      }));
+
+      let modalSpy = spyOn(ngbModal, 'open').and.callFake(() => {
+        return ({
+          componentInstance: MockNgbModalRef,
+          result: Promise.resolve({
+            isQuestionSubmitter: false,
+            isQuestionReviewer: true
+          })
+        }) as NgbModalRef;
+      });
+      component.openRoleEditor('user1');
+      tick();
```

Comment:
Add new lines above and below as it looks like this is the test "action".

Reply:
Done.